### PR TITLE
View matching

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 2.5, 3.3, 4.x
| Edition       | All

* Template rule matching every cases
* [Matchers can have multiple values](https://github.com/ibexa/core/blob/main/src/lib/MVC/Symfony/Matcher/ContentBased/MultipleValued.php#L12)
* Little grammar fix

See test in https://github.com/ibexa/documentation-developer/pull/1571#issuecomment-1708345479

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [x] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/guide/content_rendering/templates/view_matcher_reference.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ You can use the following matchers to [match content views](template_configurati
 
     You can also create [custom view matchers](custom_view_matcher.md).
 
+!!! tip
+
+    Each matcher can has a scalar value or an array of scalar values. When an array is passed, it matches on one of the values.
```
Comment (adriendupuis) [inline docs/guide/content_rendering/templates/view_matcher_reference.md:28]: TODO:
- [ ]  Validate that this is effectively true for each matcher.

**Comment by adriendupuis** [comment]
To test my example, I wrote the following files on top of a clean 4.6.0-beta1.
I do not see the interest of including those complete examples in the documentation.

* `config/packages/views.yaml`
  ```yaml
  ibexa:
      system:
          site_group:
              content_view:
                  full:
                      welcome_page:
                          template: "@ibexadesign/full/welcome_page.html.twig"
                          match:
                              Id\Location: 2
                          params:
                              project_dir: '%kernel.project_dir%'
                      no_full_view:
                          controller: App\Controller\ViewController:noFullViewAction
                          template: '@ibexadesign/full/no_full_view.html.twig'
                          match: ~
  ```
* `src/Controller/ViewController.php`
  ```php
  <?php

  namespace App\Controller;

  use Ibexa\Contracts\Core\Repository\Repository;
  use Ibexa\Contracts\Core\Repository\Values\Content\Relation;
  use Ibexa\Core\MVC\Symfony\Controller\Controller;
  use Ibexa\Core\MVC\Symfony\Security\Authorization\Attribute;
  use Ibexa\Core\MVC\Symfony\View\View;
  use Symfony\Component\HttpFoundation\Response;

  class ViewController extends Controller
  {
      public function noFullViewAction(View $view)
      {
          $view->setResponse((new Response())->setStatusCode(404));

          $allReverseRelations = $this->getRepository()->sudo(function (Repository $repository) use ($view) {
              return $repository->getContentService()->loadReverseRelations($repository->getContentService()->loadContentInfo($view->getParameter('contentId')));
          });
          $filteredReverseRelationLocations = [];
          /** @var Relation $reverseRelation */
          foreach ($allReverseRelations as $reverseRelation) {
              $content = $this->getRepository()->getContentService()->loadContent($reverseRelation->sourceContentInfo->id);
              $location = $this->getRepository()->getLocationService()->loadLocation($reverseRelation->sourceContentInfo->mainLocationId);
              if ($this->isGranted(new Attribute('content', 'read', ['valueObject' => $content, 'targets' => [$location]]))) {
                  $filteredReverseRelationLocations[] = $location;
              }
          }
          $parameters = $view->getParameters();
          $parameters['reverse_relation_locations'] = $filteredReverseRelationLocations;
          $view->setParameters($parameters);

          return $view;
      }
  }
  ```
*`templates/themes/standard/full/no_full_view.html.twig`
  ```twig
  {% extends '@ibexadesign/pagelayout.html.twig' %}

  {% block content %}
      <h1>Page not found.</h1>

      {% if 1 == reverse_relation_locations|length %}
          <p>Did you mean <a href="{{ ibexa_path(reverse_relation_locations.0) }}">{{ ibexa_content_name(reverse_relation_locations.0.contentInfo) }}</a>?</p>
      {% elseif 1 < reverse_relation_locations|length %}
          <p>Did you mean one the following?</p>
          <ul>
              {% for location in reverse_relation_locations %}
                  <li><a href="{{ ibexa_path(location) }}">{{ ibexa_content_name(location.contentInfo) }}</a></li>
              {% endfor %}
          </ul>
      {% endif %}
  {% endblock %}
  ```

File: docs/templating/templates/template_configuration.md [hunk 2]
```diff
@@ -88,5 +88,34 @@ match:
...
+    For example, you can prevent every Content item not having a dedicated template to be seen in `full` view but sent to a custom controller.
+    ```yaml
+    site_group:
+        content_view:
+            full:
+                # Rules for Content Types and specific Content items meant to be displayed in full view:
+                # …
+                # Rule for other Content items not meant to be displayed in full view:
+                no_full_view:
+                    controller: App\Controller\ViewController:noFullViewAction
```
Comment (adamwojs) [inline docs/templating/templates/template_configuration.md:117]:
```suggestion
                    controller: App\Controller\ViewController::noFullViewAction
```

Syntax with one `:` is deprecated in SF

**Review by adamwojs — APPROVED** [review]

File: docs/templating/templates/template_configuration.md [hunk 3]
```diff
@@ -88,5 +88,34 @@ match:
...
+match: ~
+match: true
+match: []
+```
+
+Such rules can be found in the [default template configuration](https://github.com/ibexa/core/blob/4.5/src/bundle/Core/Resources/config/default_settings.yml#L47).
+
+!!! tip
+
+    For example, you can prevent every Content item not having a dedicated template to be seen in `full` view but sent to a custom controller.
```
Comment (MagdalenaZuba) [inline docs/templating/templates/template_configuration.md:108]:
```suggestion
    For example, you can ensure that any Content item lacking a dedicated template isn't displayed in `full` view but is instead sent to a custom controller.
```

File: docs/templating/templates/template_configuration.md [hunk 4]
```diff
@@ -88,5 +88,34 @@ match:
...
+            full:
+                # Rules for Content Types and specific Content items meant to be displayed in full view:
+                # …
+                # Rule for other Content items not meant to be displayed in full view:
+                no_full_view:
+                    controller: App\Controller\ViewController::noFullViewAction
+                    template: '@ibexadesign/full/no_full_view.html.twig'
+                    match: ~
+    ```
+    Still for example, this custom controller could set the response status code as 404 (`$view->setResponse((new Response())->setStatusCode(404));`) and fetch some reverse relations to make suggestions in this error page.
```
Comment (MagdalenaZuba) [inline docs/templating/templates/template_configuration.md:121]:
```suggestion
    This custom controller can also set the response status code to 404 using the following code: `$view->setResponse((new Response())->setStatusCode(404));`, and fetch reverse relations to provide suggestions on the error page.
```

File: docs/templating/templates/view_matcher_reference.md [hunk 5]
```diff
@@ -32,6 +32,8 @@ You can use the following matchers to [match content views](template_configurati
 
 !!! tip
 
+    Each matcher can has a scalar value or an array of scalar values. When an array is passed, it matches on one of its values.
```
Comment (MagdalenaZuba) [inline docs/templating/templates/view_matcher_reference.md:35]:
```suggestion
    Each matcher has a scalar value or an array of scalar values. When an array is passed, it matches on one of its values.
```

**Review by MagdalenaZuba — CHANGES_REQUESTED** [review]

**Review by MagdalenaZuba — APPROVED** [review]

**Review by julitafalcondusza — APPROVED** [review]

**Review by reithor — APPROVED** [review]

File: docs/templating/templates/view_matcher_reference.md [hunk 6]
```diff
@@ -32,6 +32,8 @@ You can use the following matchers to [match content views](template_configurati
 
 !!! tip
 
+    Each matcher has a scalar value or an array of scalar values. When an array is passed, it matches on one of its values.
```
Comment (adriendupuis) [inline docs/templating/templates/view_matcher_reference.md:35]: Already there in 2.5: https://doc.ibexa.co/en/2.5/guide/content_rendering/#matcher-value

**Pull request merged**